controllers/podchaos/types.go

```diff
@@ -40,7 +45,41 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		r.Log.Error(err, "unable to get podchaos")
 		return ctrl.Result{}, nil
 	}
+	scheduler := podchaos.GetScheduler()
+	duration, err := podchaos.GetDuration()
+	if err != nil {
+		r.Log.Error(err, "unable to get duration")
+		return ctrl.Result{}, nil
+	}
+	if scheduler == nil && duration == nil {
+		return r.durationPodChaos(&podchaos, req)
+	} else if scheduler != nil && duration != nil {
+		return r.schedulePodChaos(&podchaos, req)
+	}
+
+	// This should be ensured by admission webhook in the future
+	r.Log.Error(fmt.Errorf("podchaos[%s/%s] spec invaild", podchaos.Namespace, podchaos.Name), "scheduler and duration should be omiited or defined at the same time")
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) durationPodChaos(podchaos *v1alpha1.PodChaos, req ctrl.Request) (ctrl.Result, error) {
+	switch podchaos.Spec.Action {
+	case v1alpha1.PodKillAction:
+		return r.notSupportedResponse(podchaos), nil
+	case v1alpha1.PodFailureAction:
+		r := durationPodfailure.NewDurationReconciler(r.Client, r.Log.WithValues("action", "pod-failure"), req)
+		reconciler := duration.Reconciler{
+			InnerDurationReconcile: r,
+			Client:                 r.Client,
+			Log:                    r.Log,
+		}
+		return reconciler.Reconcile(req)
```

Comment:
I think this is a little redundant, how about merge these two parts into one constructor method?